great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2542,14 +2541,15 @@ def _sqlalchemy_map_condition_query(
     unexpected_index_column_names: List[str] = result_format.get(
         "unexpected_index_column_names"
     )
-    for column_name in unexpected_index_column_names:
-        if column_name not in all_table_columns:
-            raise gx_exceptions.InvalidMetricAccessorDomainKwargsKeyError(
-                message=f'Error: The unexpected_index_column: "{column_name}" in does not exist in SQL Table. '
-                f"Please check your configuration and try again."
-            )
+    if unexpected_index_column_names:
```

Comment:
Allows for fallback behavior for `unexpected_index_query` when `unexpected_index_column_names` is not defined. The query will still be outputted, but without the ID/PK column.